src/test/java/org/assertj/core/api/AbstractAssertBaseTest.java

```diff
@@ -22,4 +27,41 @@
   protected ConcreteAssert create_assertions() {
     return new ConcreteAssert(6L);
   }
+
+  protected ConcreteAssert applyIgnoreWithoutEffects() {
+    return assertions.ignoreWhen(v -> (Long) v < 8);
+  }
+
+  protected ConcreteAssert applyIgnoreWithEffects() {
+    return assertions.ignoreWhen(v -> (Long) v > 8);
+  }
+
+  @Test
+  public void should_have_no_internal_effects_when_assertions_should_be_ignored() {
+    this.assertions = applyIgnoreWithoutEffects();
+    invoke_api_method();
+    verify_no_internal_effects();
+  }
```

Comment:
that's really nice! :+1:  it avoids us writing a test per assertions (which would be a lot of work!)

Reply:
Thanks. When I realised how many tests we would need to write, my head started spinning, so I was trying to come up with some solution 😄